File: nancykim99/pgm/python/pgm42842.py
'''
PGM42842 : 카펫 (L2)

해결 방법 : 
그냥 전체를 다 돌았다...

메모 : 
'''

def solution(brown, yellow):
    total = brown // 2 + 2 
    for h in range(3, total // 2 + 1):
        w = total - h
        if (w - 2) * (h - 2) == yellow:
            return [w, h]

File: SngJuni/PGM/pgm42842.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    for (int i = 1; i <= yellow; i++) {
        if (yellow % i != 0) continue;
        
        int w = yellow / i;
        
        int border = (w + 2) * (i + 2) - yellow;
        if (border == brown) {
            answer.push_back(w + 2);
            answer.push_back(i + 2);
            break;
        }
    }
    
    return answer;
}